### PR TITLE
Add Postgres support

### PR DESCRIPTION
**Pull request by wridgers**
- [x] Add an IRC server to `docker-compose.yml` for local testing
- [x] Actually connect to postgres
- [ ] Add a mechanism for performing migrations

### CODE REVIEW

**Comment by wridgers** [comment]
I personally prefer passing environment variables to the docker container than volume mounting a configuration file.

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+volumes:
+  pg-data:
+
+services:
+  jeeves:
+    build: .
+    depends_on:
+      - db
+      - irc
+    environment:
```
Comment (nullren) [inline docker-compose.yml]: i really only preferred config files because you don't need to rewrite the run/launch scripts to run the bot

**Comment by wridgers** [comment]
@nullren I've done the most I'm probably gonna do today if you fancy finishing it off

**Comment by nullren** [comment]
I just want to make sure everything works independently of each other, for example being able to run from npm directly, run from docker, and run from docker compose.

Also, being able to swap out psql for anything else would probably be a good thing since postgres is really overkill for our use lol. Like we should not have any problems swapping out psql for sqlite or even probably just serializing a json list to a file.

I didn't do it in this last commit, but I'm probably going to generalize the config a little bit so we can use environment variables or config files. I would like to use jeeves separately and just run it easily from npm.

**Comment by nullren** [comment]
updated travis to build and run tests from docker-compose

File: docker-compose.yml (hunk 1 above)
Comment (wridgers) [reply docker-compose.yml]: sure. env vars are good for docker, config files are good for npm run.

**Comment by wridgers** [comment]
nice! we could probably add an integration test for being able to connect to irc too

**Comment by wridgers** [comment]
Not sure how the migration code works. afaiu it should work like this:

  - Create `migrations` table if it doesn't exist
  - Look in `migrations/` for an ordered set of database schema changes
  - For each `migrations/<migration>.sql`
    - If `<migration>` is in `migrations` table, continue
    - Execute contents of `<migration>.sql`
    - Add `<migration>` to `migrations` table

This way there is a mechanism to make changes to the database schema (add tables/columns, drop columns, etc).

**Comment by wridgers** [comment]
> Also, being able to swap out psql for anything else would probably be a good thing since postgres is really overkill for our use lol. Like we should not have any problems swapping out psql for sqlite or even probably just serializing a json list to a file.

Why would we ever do this? Docker makes it really easy to run postgres, and being able to run real SQL is too useful.

**Comment by nullren** [comment]
Is this pr... alive?

**Pull request merged**